stripe/src/main/java/com/stripe/android/model/Card.java

```diff
@@ -138,54 +189,89 @@ public Builder currency(String currency)
+         * @return
+         */
         public Card build() {
             return new Card(this);
         }
     }
 
-    private Card(Builder builder) {
-        this.number = TextUtils.nullIfBlank(normalizeCardNumber(builder.number));
-        this.expMonth = builder.expMonth;
-        this.expYear = builder.expYear;
-        this.cvc = TextUtils.nullIfBlank(builder.cvc);
-        this.name = TextUtils.nullIfBlank(builder.name);
-        this.addressLine1 = TextUtils.nullIfBlank(builder.addressLine1);
-        this.addressLine2 = TextUtils.nullIfBlank(builder.addressLine2);
-        this.addressCity = TextUtils.nullIfBlank(builder.addressCity);
-        this.addressState = TextUtils.nullIfBlank(builder.addressState);
-        this.addressZip = TextUtils.nullIfBlank(builder.addressZip);
-        this.addressCountry = TextUtils.nullIfBlank(builder.addressCountry);
-        this.last4 = TextUtils.nullIfBlank(builder.last4);
-        this.type = TextUtils.nullIfBlank(builder.type);
-        this.fingerprint = TextUtils.nullIfBlank(builder.fingerprint);
-        this.country = TextUtils.nullIfBlank(builder.country);
-        this.type = getType();
-        this.last4 = getLast4();
-        this.currency = TextUtils.nullIfBlank(builder.currency);
-    }
-
-    public Card(String number, Integer expMonth, Integer expYear, String cvc, String name, String addressLine1, String addressLine2, String addressCity,
-                String addressState, String addressZip, String addressCountry, String last4, String type, String fingerprint, String country) {
-        this.number = TextUtils.nullIfBlank(normalizeCardNumber(number));
+    /**
+     * Card constructor with all available fields.
+     *
+     * @param number the credit card number
+     * @param expMonth the expiry month
+     * @param expYear the expiry year
+     * @param cvc the CVC number
+     * @param name the card name
+     * @param addressLine1 first line of the billing address
+     * @param addressLine2 second line of the billing address
+     * @param addressCity city of the billing address
+     * @param addressState state of the billing address
+     * @param addressZip zip code of the billing address
+     * @param addressCountry country for the billing address
+     * @param last4 last 4 digits of the card
+     * @param fundingType the funding type of this card
+     * @param fingerprint the card fingerprint
+     * @param country ISO country code of the card itself
+     * @param currency currency used by the card
+     */
+    public Card(
+            String number,
+            Integer expMonth,
+            Integer expYear,
+            String cvc,
+            String name,
+            String addressLine1,
+            String addressLine2,
+            String addressCity,
+            String addressState,
+            String addressZip,
+            String addressCountry,
+            @Size(4) String last4,
```

Comment:
I notice that I'm mixing my logic on whether or not to allow the constructor to set the brand and last4 or to determine it from the number. I should allow the determined values to be overwritten (in case we're creating this model without knowing the full card number). Will update.